pyccel/ast/cwrapper.py

```diff
@@ -119,3 +119,58 @@ def flags(self):
     @property
     def args(self):
         return self._result_args
+
+class CastFunction(FunctionDef):
+    """Represents a cast function definition."""
+
+    def __init__(self,
+        name,
+        cast_type,
+        arguments,
+        ret,
+        results):
+        self._name = name
+        self._arguments = arguments
+        self._ret = ret
+        self._results = results
+        self._cast_type = cast_type
+        body_ = ''
+        #TODO I dont know if the build of body shoudl be done here or in the cwrapper
+        #TODO this is just a tmp way of printing this shoudlbe improved later
+        if self._cast_type == 'pyint_to_bool':
+            body_ += '{} = {} != 0;\n'.format(self.results[0].name, self._arguments[0].name)
+
+        elif self.cast_type == 'bool_to_pyobj': 
+            body_ += '{} = {} != 0 ? Py_True : Py_False;\n'.format(self.results[0].name, self._arguments[0].name)
+        self._body = body_
```

Comment:
I think these should be instances of the `FunctionDef` class rather than being a subclass. As I see it, the reasons to use a class are:
1. To add some special logic in the initialisation (here the logic would be the same if this were an instance rather than a class)
2. To benefit from a common functionality which should be handled in the same way for multiple things (here the common functionality is all entirely described by the `FunctionDef` class)
3. To avoid code repetition (each if option will only be called once so there is no repetition eliminated)

I don't see that we have gained any of these advantages

Reply:
I used it as a sub class   so i can add __hash__   __eq__ methods  and because  I didn't know how  things will be later on the body part .

Reply:
Ah ok. If we use a dictionary then do we still need the `__hash__` and `__eq__` methods?
Even if the body changes it will still be unique to each conversion won't it?

Reply:
if we use dict we can  forget about __hash__  and __eq__    bqz we can use the cast_type (string) as a key